pdr_backend/subgraph/subgraph_payout.py

```diff
@@ -0,0 +1,143 @@
+        skip: The number of records to skip (pagination offset).
+
+    Returns:
+        A string representing the GraphQL query.
+    """
+
+    # asset_ids_str = str(asset_ids).replace("[", "[").replace("]", "]").replace("'", '"')
+    where_query_arr = []
+
+    for asset_id in asset_ids:
+        where_query_arr.append(
+            """
+                {
+                    timestamp_gte: %s,
+                    timestamp_lte: %s,
+                    prediction_contains: "%s"
+                }
+            """
+            % (start_ts, end_ts, asset_id)
+        )
+
+    return """
+        query {
+            predictPayouts (
+            first: %s
+            skip: %s
+            where: {
+                or: [%s]
+            }
+        ) {
+                id
+                timestamp
+                payout
+                prediction {
+                    user {
+                        id
+                    }
+                    slot {
+                        id
+                        predictContract{
+                            id
+                            token{
+                                name
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    """ % (
+        first,
+        skip,
+        ", ".join(where_query_arr),
+    )
+
+
+@enforce_types
+def filter_by_addresses(result, addresses):
+    """
+    Filter the result["data"]["predictPayouts"] by a list of addresses.
+
+    Parameters:
+    result (dict): The result dictionary.
+    addresses (list): The list of addresses to filter by.
+
+    Returns:
+    list: The filtered list of payouts.
+    """
+    payouts = result["data"]["predictPayouts"]
+    filtered_payouts = [
+        payout
+        for payout in payouts
+        if payout["prediction"]["slot"]["predictContract"]["id"] in addresses
+    ]
+
+    return filtered_payouts
+
+
+@enforce_types
+def fetch_payouts(
+    addresses: List[str],
+    start_ts: int,
+    end_ts: int,
+    skip: int,
+    network: str = "mainnet",
+) -> List[Payout]:
+    records_per_page = 10
+    query = get_payout_query(
+        addresses,
+        start_ts,
+        end_ts,
+        records_per_page,
+        skip,
+    )
+
+    print("payout query", query)
+
+    result = query_subgraph(
+        get_subgraph_url(network),
+        query,
+        timeout=20.0,
+    )
+
+    new_payouts = filter_by_addresses(result, addresses)
+
+    if new_payouts is None:
+        return []
+
+    new_payouts = [
+        Payout(
+            **{
+                "payout": float(payout["payout"]),
+                "user": payout["prediction"]["user"]["id"],
+                "timestamp": payout["timestamp"],
+                "ID": payout["id"],
+                "token": payout["prediction"]["slot"]["predictContract"]["token"][
+                    "name"
+                ],
+                "slot": int(payout["id"].split("-")[1]),
```

Comment:
Please make sure that we're retrieving all required data to generate the insights we want.

Example: Payout is where [predictedValue + prediction.slot.revenue + prediction.stake + prediction.slot.roundSumStakesUp + prediction.slot.roundSumStakes] are provided. So, we need to fetch this additional data such that we can update our local records.

We can then use the data from the payout event to update our lake/ `pdr_predictions` and `pdr_slots` tables.

```
query{  
  predictPayouts(...){
    prediction{
      slot {
        id
        predictContract {
          id
        }
        slot
        status
        revenue
        roundSumStakesUp
        roundSumStakes
      }
      user {
        id
      }
      stake
    }
    payout
    predictedValue
    trueValue
    timestamp
  }
}```

Reply:
https://github.com/oceanprotocol/pdr-backend/issues/447
I have followed the BRONZE_FOR_PAYOUTS table on the epic so I didn't add these values.

I am going to add them asap.